src/realm/integer_flex_compressor.hpp

```diff
@@ -189,116 +190,99 @@ inline bool FlexCompressor::find_all(const Array& arr, int64_t value, size_t sta
-            return item > key;
-        REALM_UNREACHABLE();
-    };
-
-    const auto& c = arr.integer_compressor();
-    const auto offset = c.v_width() * c.v_size();
-    const auto ndx_w = c.ndx_width();
-    const auto v_w = c.v_width();
-    const auto data = c.data();
-    const auto mask = c.v_mask();
-    BfIterator ndx_iterator{data, offset, ndx_w, ndx_w, start};
-    BfIterator data_iterator{data, 0, v_w, v_w, static_cast<size_t>(*ndx_iterator)};
-    while (start < end) {
-        const auto sv = sign_extend_field_by_mask(mask, *data_iterator);
-        if (cmp(sv, value) && !state->match(start + baseindex))
-            return false;
-        ndx_iterator.move(++start);
-        data_iterator.move(static_cast<size_t>(*ndx_iterator));
-    }
-    return true;
-}
-
-template <typename VectorCond1, typename VectorCond2>
-inline bool FlexCompressor::find_parallel(const Array& arr, int64_t value, size_t start, size_t end, size_t baseindex,
-                                          QueryStateBase* state)
-{
-    //
-    // algorithm idea: first try to find in the array of values (should be shorter in size but more bits) using
-    // VectorCond1.
-    //                 Then match the index found in the array of indices using VectorCond2
-    //
-
     const auto& compressor = arr.integer_compressor();
-    const auto v_width = compressor.v_width();
+    const auto v_width = arr.m_width;
     const auto v_size = compressor.v_size();
-    const auto ndx_width = compressor.ndx_width();
-    const auto offset = v_size * v_width;
+    const auto mask = compressor.v_mask();
     uint64_t* data = (uint64_t*)arr.m_data;
+    size_t v_start = realm::not_found;
 
-    auto MSBs = compressor.msb();
-    auto search_vector = populate(v_width, value);
-    auto v_start =
-        parallel_subword_find(find_all_fields<VectorCond1>, data, 0, v_width, MSBs, search_vector, 0, v_size);
+    {
+        int64_t v = value;
+        if constexpr (std::is_same_v<Cond, Greater>) {
+            v++; // We use GreaterEqual below, so this will effectively be Greater
+        }
 
-    if constexpr (!std::is_same_v<VectorCond2, NotEqual>) {
-        if (start == v_size)
-            return true;
+        if (v_width <= 16 && v_size >= 20) {
+            auto search_vector = populate(v_width, v);
+            v_start = parallel_subword_find(find_all_fields<GreaterEqual>, data, 0, v_width, compressor.msb(),
+                                            search_vector, 0, v_size);
+        }
+        else {
+            BfIterator data_iterator{data, 0, v_width, v_width, 0};
+            size_t idx = 0;
+            while (idx < v_size) {
+                if (sign_extend_field_by_mask(mask, *data_iterator) >= v) {
+                    break;
+                }
+                data_iterator.move(++idx);
+            }
+            v_start = idx;
+            /*
+            // Binary search alternative
+            v_start = lower_bound(v_size, v, mask, data_iterator);
```

Comment:
Remove if we don't intend to pursue this.

Reply:
Will do